src/proc/terminator.rs

```diff
@@ -0,0 +1,40 @@
+/// Ensure that the given block has return statements
+/// at the end of its control flow.
+///
+/// Note: we don't want to blindly append a return statement
+/// to the end, because it may be either redundant or invalid,
+/// e.g. when the user already has returns in if/else branches.
+pub fn ensure_block_returns(block: &mut crate::Block) {
+    match block.last_mut() {
+        Some(&mut crate::Statement::Block(ref mut b)) => {
+            ensure_block_returns(b);
+        }
+        Some(&mut crate::Statement::If {
```

Comment:
Why do we need to add it on both sides of the if? It seems it could be sufficient to add a single return at the end of `block` (and maybe it is unreachable). It would make this function much simpler because it would just check if the last statement is a `Return` and if not, then add one.

Reply:
I'm not totally sure if that way is better, but the reasoning is expressed at the top of the function:
>/// Note: we don't want to blindly append a return statement
/// to the end, because it may be either redundant or invalid,
/// e.g. when the user already has returns in if/else branches.

Reply:
I.e. we don't want to generate IR that looks like this: 
```
if xxx { return; } else { return; }
return
```
Because then we'd hit our own validation error that there is stuff after `return`.

Reply:
Ack, LGTM then!